test/torchaudio_unittest/transforms/autograd_test_impl.py

```diff
@@ -0,0 +1,45 @@
+from parameterized import parameterized
+from torch.autograd import gradcheck, gradgradcheck
+import torchaudio.transforms as T
+
+from torchaudio_unittest.common_utils import (
+    TestBaseMixin,
+    get_whitenoise,
+)
+
+
+class AutogradTestCase(TestBaseMixin):
+    def assert_grad(self, transform, *inputs, eps=1e-06, atol=1e-05, rtol=0.001):
+        transform = transform.to(self.device, self.dtype)
+
+        inputs_ = []
+        for i in inputs:
+            i.requires_grad = True
+            inputs_.append(i.to(dtype=self.dtype, device=self.device))
+        assert gradcheck(transform, inputs_, eps=eps, atol=atol, rtol=rtol)
+        assert gradgradcheck(transform, inputs_, eps=eps, atol=atol, rtol=rtol)
+
+    @parameterized.expand([
+        ({'pad': 0, 'normalized': False, 'power': None}, ),
+        ({'pad': 3, 'normalized': False, 'power': None}, ),
+        ({'pad': 0, 'normalized': True, 'power': None}, ),
+        ({'pad': 3, 'normalized': True, 'power': None}, ),
+        ({'pad': 0, 'normalized': False, 'power': 1.0}, ),
+        ({'pad': 3, 'normalized': False, 'power': 1.0}, ),
+        ({'pad': 0, 'normalized': True, 'power': 1.0}, ),
+        ({'pad': 3, 'normalized': True, 'power': 1.0}, ),
+        ({'pad': 0, 'normalized': False, 'power': 2.0}, ),
+        ({'pad': 3, 'normalized': False, 'power': 2.0}, ),
+        ({'pad': 0, 'normalized': True, 'power': 2.0}, ),
+        ({'pad': 3, 'normalized': True, 'power': 2.0}, ),
+    ])
+    def test_spectrogram(self, kwargs):
+        transform = T.Spectrogram(**kwargs)
+        waveform = get_whitenoise(sample_rate=8000, duration=0.05, n_channels=2)
```

Comment:
I think you're missing the device/dtype args here no?
That might be simpler than moving stuff around in the assert_grad function.

Reply:
This is the pattern we use in `torchaudio`, where we define the tests in base class and set the `device/dtype` in separate class.
Also to minimize the chance to write inconsistent test, the conversion to the expected `device/dtype` is abstracted in helper functions.
Plus, since this tests only run on float64, I will stop parametrizing the `dtype`. will update the code soon